### ISSUE TITLE
NullReferenceException in CheckAndAddArrayVersionLog

### ISSUE THREAD

**Issue by chrhol** [Starcounter/Starcounter#4001]
Concerns version 2.3 and develop.

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Starcounter.Json.CheckAndAddArrayVersionLog(ChangeLog clog) in C:\TeamCity\BuildAgent\work\sc-78\Level1\src\Starcounter.XSON\Class Json\Json.Array.cs:line 117
   at Starcounter.Templates.TObjArr.CheckAndSetBoundValue(Json parent, Boolean addToChangeLog) in C:\TeamCity\BuildAgent\work\sc-78\Level1\src\Starcounter.XSON\Templates\TObjArr.cs:line 166
   at Starcounter.Json.<>c__DisplayClass125_0.<SetBoundValuesInTuple>b__0(Json json) in C:\TeamCity\BuildAgent\work\sc-78\Level1\src\Starcounter.XSON\Class Json\Json.Object.ChangeLog.cs:line 365
   at Starcounter.Advanced.XSON.JsonExtension.Scope[T](Json json, Action`1 action, T arg) in C:\TeamCity\BuildAgent\work\sc-78\Level1\src\Starcounter.XSON\Class Json\JsonExtension.cs:line 95
   at Starcounter.Json.CallHasAddedElement(Int32 index, Json item) in C:\TeamCity\BuildAgent\work\sc-78\Level1\src\Starcounter.XSON\Class Json\Json.Object.GetSetValue.cs:line 241
   at Starcounter.Json._Add(Object item) in C:\TeamCity\BuildAgent\work\sc-78\Level1\src\Starcounter.XSON\Class Json\Json.IList.cs:line 259
   at Launcher.LauncherHelper.WrapInWorkspace(Request request, Json resource) in C:\GitHub\Palma\PalmaServer\Launcher\src\Launcher\Api\LauncherHelper.cs:line 102
   at Launcher.LauncherHelper.<Register>b__1_1(Request request, Response response) in C:\GitHub\Palma\PalmaServer\Launcher\src\Launcher\Api\LauncherHelper.cs:line 38
   at Starcounter.Handle.RunResponseFilters(Request req, Response resp) in C:\TeamCity\BuildAgent\work\sc-78\Level1\src\Starcounter.Internal\Handle.cs:line 550
   at Starcounter.Request.SendResponse(Response resp, Byte[] serializationBuf) in C:\TeamCity\BuildAgent\work\sc-78\Level1\src\Starcounter.Internal\Http\Request.cs:line 1211
   at Starcounter.Internal.AppsBootstrapper.ProcessExternalRequest(Request req) in C:\TeamCity\BuildAgent\work\sc-78\Level1\src\Starcounter.Apps.JsonPatch\AppsBootstrapper.cs:line 811
HResult=-2147467261

More information about this error may be available in the server error log.
```

There are two problems: not checking if changelog is null and that this method is called when it shouldn't be.

**Comment by chrhol** [comment]
This issue is already fixed in branch `issue/v2.3/4001`, but since teamcity is not up I haven't pushed it yet.

**Comment by chrhol** [comment]
Pushed to `v2.3` and `develop` branches now.
